packages/core-data/src/selectors.js

```diff
@@ -180,3 +193,42 @@ export function isPreviewEmbedFallback( state, url ) {
 export function hasUploadPermissions( state ) {
 	return state.hasUploadPermissions;
 }
+
+/**
+ * Returns the autosave that is a child of the provided post, if one exists.
+ *
+ * @param {Object} state    State tree.
+ * @param {string} postType The type of the parent post.
+ * @param {number} postId   The id of the parent post.
+ *
+ * @return {?Object} The autosave object, or undefined if there is none.
+ */
+export function getAutosave( state, postType, postId ) {
+	return state.autosave[ postId ];
+}
+
+/**
+ * Returns the true if there is an autosave for the given post, otherwise false.
+ *
+ * @param {Object} state State tree.
+ * @param {string} postType The type of the parent post.
+ * @param {number} postId   The id of the parent post.
+ *
+ * @return {boolean} Whether there is an existing autosave.
+ */
+export function hasAutosave( state, postType, postId ) {
+	return !! getAutosave( state, postType, postId );
```

Comment:
Accessing another selector in the same file like this won't trigger the resolver, which makes it incorrectly return false when the autosave might simply not have been fetched yet. I'm wondering what the best alternative would be ... maybe just drop this selector in favour of `!! select( 'core' ).getAutosave()`?

Reply:
I decided against adding `hasAutosave` to core data, like `getAutosaveAttribute` it wasn't being used after my changes. Deleting it was the simplest solution!